extensions/indexes/sort/src/main/java/org/exist/xquery/modules/sort/CreateOrderIndex.java

```diff
@@ -104,11 +104,11 @@ public Sequence eval(final Sequence[] args, final Sequence contextSequence) thro
             final Element options = (Element) optionValue.getNode();
             String option = options.getAttribute("order");
             if (option != null) {
-                descending = option.equalsIgnoreCase("descending");
+                descending = "descending".equalsIgnoreCase(option);
             }
             option = options.getAttribute("empty");
             if (option != null) {
-                emptyLeast = option.equalsIgnoreCase("least");
+                emptyLeast = "least".equalsIgnoreCase(option);
             }
```

Comment:
May could be written even simpler like so:
```
            descending = "descending".equalsIgnoreCase(options.getAttribute("order"));
            emptyLeast = "least".equalsIgnoreCase(options.getAttribute("empty"));
```

Reply:
Careful, we should make sure that option can not be null. I saw tests fail even though IntelliJ was sure the attributes and elements can never be null, they can in our codebase.

Reply:
Nevermind, this works because the check afterwards handles null as well.